.github/workflows/compliance.yml

```diff
@@ -0,0 +1,114 @@
+#                  or uses the SfdoInstrumentationService. This is primarily meant as a short term action to aid in
+#                  adoption of the new instrumentation services.
+#####################################################################################################################
+name: Instrumentation_Service_Compliance_Verification
+on:
+  pull_request:
+    ## Filter the job to only execute if CLS files are in the PR
+    paths:    
+      - '**.cls'
+      - '!unpackaged/**.cls'
+jobs:
+  Code_Check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Check for previously assigned Instrumentation Label
+        id: label_check
+        run: |
+          echo "::set-output name=prlabels::${{ join(github.event.pull_request.labels.*.name, ',') }}"
+
+      - name: Check for references to the SfdoLogUtils or SfdoInstrumentationServices Apex classes
+        # If the Instrumentation label is already assigned to the PR, all checks are skipped to avoid redundant PR comments and alerts
+        id: get_pr_diff
+        if: contains(steps.label_check.outputs.prlabels, 'Instrumentation') == false
+        run: |
+          echo "Did not find label 'Instrumentation' in '${{steps.label_check.outputs.prlabels}}'"
+          if [ $GITHUB_BASE_REF ]; then
+            # This changes the context of subseqent commands to get the full change for the PR
+            git fetch origin $GITHUB_HEAD_REF:$GITHUB_HEAD_REF
+
+            # Find references to "Sfdo" in the force-app/main/ folder. Filters for the specific class references in later steps.
+            export DIFF=$( git diff-tree origin/$GITHUB_BASE_REF..$GITHUB_HEAD_REF --patch-with-raw -- force-app/main/** | grep 'Sfdo' )
+          fi
+
+          # Assign the "git diff-tree" output to an internal workflow var to use later
+          # Escape newlines (replace \n with %0A) so it's a single line of text
+          echo "::set-output name=diff::$( echo "$DIFF" | sed ':a;N;$!ba;s/\n/%0A/g' )"
+
+          # echo "Diff Output = ${{ steps.get_pr_diff.outputs.diff }}"
+
+      - name: Assign Reviewer if SfdoInstrumentationService is in use
+        id: assign_reviewers_if_instrumentation
+        if: contains(steps.get_pr_diff.outputs.diff, 'SfdoInstrumentationService')
+        uses: SalesforceFoundation/github-script@v4
+        with:
+          script: |
+            github.issues.addLabels({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+              labels: [ "Instrumentation" ]
+            })
+            github.issues.addAssignees({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+              assignees: [ "force2b" ]
```

Comment:
Suggested `TODO`: You can improve maintainability here by moving `force2b` to a workflow-level environment variable.

Reply:
@jstvz  I was wondering about that.  Is the right syntax something like:
`echo "::set-output name=reviewer::force2b"`?

What would be even more interesting is to dynamically select the reviewer/assignee based on the owner of the PR. If it's owned by _me_, I can't be set as the reviewer (which is why you're assigned below). So maybe the line above can dynamically change the reviewer based on the current PR owner.  Not quite sure how to do that yet, but I'm somewhat assuming that the owner is in the context possibly.

Reply:
> I was wondering about that. Is the right syntax something like:
echo "::set-output name=reviewer::force2b"?

I'd use [`env`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#env) here. You could access it with `process.env.VAR` in the script.

> What would be even more interesting is to dynamically select the reviewer/assignee based on the owner of the PR. 

I see a couple of options:

1. Request review from a team... that may or may not work if you're the only member.
2. Add some logic to select the user based on the value of `context.payload.pull_request.user.login`